lib/takeaway.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'menu'
+
+class Takeaway
+  include Menu
+
+  def initialize
+    @final_order = []
+    @hashed_order = {}
+    @takeaway_menu = { "pizza" => 3, "curry" => 4, "chopsuey" => 2 }
+  end
+
+  attr_reader :final_order, :takeaway_menu, :hashed_order
+
+  def order(dish, qty)
+    fail 'That is not on the menu!' unless dish == "curry" or dish == "chopsuey" or dish == "pizza"
+    final_order << [dish, qty]
```

Comment:
You are not currenlty checkint that any of these items are actually on the menu - what happens if the menu changes and no longer has chopsuey? Your fail clause would no longer be accurate.